my_enumerables.rb

```diff
@@ -0,0 +1,203 @@
+module Enumerable
+  def my_each
+    modified_object = []
+
+    i = 0
+    if block_given?
+      while i < length
+        modified_object << (yield self[i])
+        i += 1
+      end
+      self
+    else
+      to_enum
+    end
+  end
+
+  def my_each_with_index
+    modified_object = []
+
+    i = 0
+    if block_given?
+      while i < to_a.length
+        modified_object << (yield self[i], i)
+        i += 1
+      end
+      self
+    else
+      to_enum
+    end
+  end
+
+  def my_select
+    modified_object = []
+
+    i = 0
+    if block_given?
+      while i < length
+        modified_object << self[i] if (yield self[i]) == true
+        i += 1
+      end
+      modified_object
+    else
+      to_enum
+    end
+  end
+
+  def my_all?(pattern = nil)
+    true if length.zero?
+    selected_array = []
+
+    i = 0
+    if block_given?
+      selected_array = my_select { |element| yield element }
+    elsif !block_given? && pattern
+      while i < length
+        selected_array << self[i] if pattern === self[i]
+        i += 1
+      end
+    elsif !block_given? && !pattern
+      while i < length
+        selected_array << self[i] if self[i]
+        i += 1
+      end
+    end
+    selected_array.length == length
+  end
+
+  def my_any?(pattern = nil)
+    false if length.zero?
+    selected_array = []
+
+    i = 0
+    if block_given?
+      selected_array = my_select { |element| yield element }
+    elsif !block_given? && pattern
+      while i < length
+        selected_array << self[i] if pattern === self[i]
+        i += 1
+      end
+    elsif !block_given? && !pattern
+      while i < length
+        selected_array << self[i] if self[i]
+        i += 1
+      end
+    end
+    selected_array.length.positive?
+  end
+
+  def my_none?(pattern = nil)
+    true if length.zero?
+    selected_array = []
+
+    i = 0
+    if block_given?
+      selected_array = my_select { |element| yield element }
+    elsif !block_given? && pattern
+      while i < length
+        selected_array << self[i] if pattern === self[i]
+        i += 1
+      end
+    elsif !block_given? && !pattern
+      while i < length
+        selected_array << self[i] if self[i]
+        i += 1
+      end
+    end
+    selected_array.length.zero?
+  end
+
+  def my_count(*args)
+    count_array = []
+
+    i = 0
+    if block_given?
+      while i < length
+        count_array << self[i] if yield self[i]
+        i += 1
+      end
+    elsif !block_given? && !args.empty?
+      while i < length
+        count_array << self[i] if self[i] == args.first
+        i += 1
+      end
+    else
+      count_array = self
+    end
+    count_array.length
+  end
```

Comment:
`my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))
the count can also perform with range too. But your one cannot perform with range. While I was playing with your method just came up with a formation that works. But need a lot of refactoring. Or you can generate your own form of solution. The issue I am facing while executing my_count with range
```ruby
(1...3).my_count # should return 2
(1...3).my_count(2) # should return 1
```
Now the solution that I have generated by playing with your code.
```ruby
def my_count(*args)
    arr = is_a?(Array) ? self : to_a
    count = 0
    i = 0
    if block_given?
      while i < arr.length
        count += 1 if yield arr[i]
        i += 1
      end
    elsif !block_given? && !args.empty?
      while i < arr.length
        count += 1 if arr[i] === args.first
        i += 1
      end
    else
      while i < arr.length
        count += 1 if arr[i]
        i += 1
      end
    end
    count
  end
```